app/src/main/kotlin/com/exallium/todoapp/entities/Note.kt

```diff
@@ -3,20 +3,14 @@ package com.exallium.todoapp.entities
 /**
  * The Note entity
  */
-data class Note(val id: String,
-                val title: String,
-                val body: String,
-                val created: Long,
-                val updated: Long) {
+data class Note(val id: String = EMPTY_STRING,
+                val title: String = EMPTY_STRING,
+                val body: String = EMPTY_STRING,
+                val created: Long = EMPTY_TIMESTAMP,
+                val updated: Long = EMPTY_TIMESTAMP) {
 
-    fun with(id: String? = null,
-             title: String? = null,
-             body: String? = null,
-             created: Long? = null,
-             updated: Long? = null) = Note(
-            id ?: this.id,
-            title ?: this.title,
-            body ?: this.body,
-            created ?: this.created,
-            updated ?: this.updated)
+    companion object {
+        val EMPTY_STRING = ""
+        val EMPTY_TIMESTAMP = 0L
```

Comment:
so the current validation i have checks if the field is not just empty but blank which is better than comparing to Note.EMPTY_STRING. unless i add several specific validation methods on Note all checking the same thing (empty string), my current validation is fine and the empty fields in the constructor are only used for something like firebase that requires no args constructor.